lib/Raven/Client.php

```diff
@@ -233,35 +238,49 @@ class Client
      */
     protected $_shutdown_function_has_been_set;
 
-    public function __construct($options_or_dsn = null, $options = [])
+    /**
+     * @var Configuration The client configuration
+     */
+    protected $config;
+
+    /**
+     * Constructor.
+     *
+     * @param Configuration $config The client configuration
+     */
+    public function __construct(Configuration $config)
```

Comment:
Because in some cases instantiating the transport could create a connection, that could be cumbersome for resources, or simply being too early, could timeout, or even be not needed since no event will be fired to Sentry.